### DISCUSSION TITLE
解决内联开销没有去掉压栈操作，能否继续优化？

### DISCUSSION THREAD

**Discussion by pengkou** [bytedance/sonic#322] [Q&A]
关于golang不能内联，文档中提到使用了轻量级的函数调用

> Global-function-table + static offset for calling instruction
> Pass parameters using registers

但是好像并没有解决不能内联引入的压栈出栈开销（_Assembler中的prologue，epilogue）

请问这里是否可以继续优化？若有不对，请指正。

**Comment by AsterDY** [comment]
目前还不能，go internal ABI 还不对ASM开放。后续会考虑使用类似 DLL 的方式动态链接 native 函数绕过 go asm 的限制

**Reply by pengkou** [reply]
动态链接库中的函数能内联吗？

**Reply by AsterDY** [reply]
这个 PR #393  实现了 register-based ABI for asm2asm
